plugins/scripting/scripting-compiler/src/org/jetbrains/kotlin/scripting/compiler/plugin/repl/ReplCodeAnalyzer.kt

```diff
@@ -43,13 +46,13 @@ class ReplCodeAnalyzer(environment: KotlinCoreEnvironment) {
 
     val module: ModuleDescriptorImpl
 
-    val trace: BindingTraceContext = NoScopeRecordCliBindingTrace()
+    val trace: BindingTraceContext = CliBindingTrace()
```

Comment:
Making separate trace is rather complicated in the current API, because it's binded to the compilation state which is shared over all compilations, and spiltting it seems to be not too easy. I hope we'll also discuss it with @nikitinas